feat(docs): Improve documentation on filtering values. (#213)
someengineering/resoto.com: docs/concepts/search/filters.md

The `property_path` is the path to the property in the JSON structure.

A nested property can be accessed by defining the path in the object structure separated by dot (`.`). The path `some.nested.prop` points to 42 in this json structure:

```json
{ "some": { "nested": { "prop": 42 } } }
```

In case a part of the path is not a valid json path, it can be escaped with backticks `` some.`non.compliant.key`.prop ``.

```json
{ "some": { "non.compliant.key": { "prop": 42 } } }
```

> In case the nested property is an array, we are able to access the specific element by using the index notation.
> 
> ```json
> { "some": { "nested": [1, 2, 3] } }
> ```
> 
> We can use the path `some.nested[0]` to access the first element of the array, and `some.nested[2]` to access the third element.
> 
> If the array contains objects, we can access the specific element by using the index notation: `some.nested[0].prop`.
> 
> ```json
> { "some": { "nested": [{ "prop": 1 }, { "prop": 2 }, { "prop": 3 }] } }
> ```

Reviewer: ````suggestion
#### Arrays

If a property is an array, specific elements can be accessed using index notation.

In the below JSON, the path `some.nested[0]` references first element of the array with path `some.nested`, and `some.nested[2]` points to the third (and last) element:

```json
{ "some": { "nested": [1, 2, 3] } }
```

In arrays of objects, object elements can also be accessed using index notation.

`some.nested[0].prop` has a value of `1` here:

```json
{ "some": { "nested": [{ "prop": 1 }, { "prop": 2 }, { "prop": 3 }] } }
```
````

> In order to select elements, where the first nested prop is bigger than 2, we can use the path `some.nested[0].prop > 2`. If we want to select elements that have any nested prop bigger than 2, we can use the path `some.nested[*].prop > 2`. The `*` wildcard is used to perform the filter criteria on all elements of the array, where only one match would be sufficient to match.

Reviewer: I don't quite understand this portion. Could you clarify the meaning or give an example?

### Property sections

> Most of the properties are defined in the reported section. The CLI interprets all defined property paths relative to the reported section by default. Thus, the path to property `reported.name` can simply be written as `name`. When every path is interpreted relative to `reported`, we need a way to target properties not in the reported section. This is possible by using the absolute path syntax via the `/` (slash). A property path that starts with a slash is always interpreted absolute.
> 
> In order to access properties outside of the reported section, use the `/` syntax:

Reviewer: ```suggestion
Most properties are defined in the `reported` section, and the CLI interprets property paths relative to the `reported` section by default. Thus, the path to property `reported.name` can simply be written as `name`.

To target properties not in the `reported` section, prefix the property path with a slash (`/`) for it to be interpreted as an absolute path:
```
